.github/workflows/ci_linux.yml

```
@@ -0,0 +1,120 @@
+name: CI Linux
+
+on:
+  pull_request:
+    branches:
+      - '**'
+  workflow_dispatch:
+    inputs:
+      name:
+        description: "Manual trigger"
+
+jobs:
+  build-test-style-cover:
+    name: Build, Tests, and Coverage
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

I think we need to specify here the branch to checkout for PR-triggered workflows. Otherwise, we are always compiling against the default repo's branch.


```suggestion
      - name: Checkout
        uses: actions/checkout@v5
        with:
          fetch-depth: 0
          ref: ${{ github.ref }}
```

---

Addressed in 4117a4c9db7b0e210769cdf71a1d3ad7bc28e809